torch/_higher_order_ops/utils.py

```diff
@@ -313,16 +313,23 @@ def fw_with_masks(*args):
 # In PyTorch there is the convention that None gradients are
 # replaced with all-zeros gradients
 def unmask_none_gradients(grads, operands):
-    return [
-        g
-        if (
-            (isinstance(o, torch.Tensor) and not o.requires_grad)
-            or not isinstance(o, torch.Tensor)
-            or g is not None
-        )
-        else torch.zeros_like(o)
-        for g, o in zip(grads, operands)
-    ]
+    allowed_types = (torch.Tensor, int, torch.SymInt)
+    unmasked_grads = []
+    for g, o in zip(grads, operands):
+        if not isinstance(o, allowed_types) or g is not None:
```

Comment:
there shouldn't be any o that's not in allowed types, so we could just error out if that's the case.

Reply:
I have changed this. Now if any operands is of a different types as torch.Tensor, int or torch.SymInt, we error out.